src/client/entity-editor/relationship-editor/relationship-editor.js

```diff
@@ -0,0 +1,339 @@
+
+	handleEntityChange = (value: EntitySearchResult) => {
+		this.setState({
+			relationship: null,
+			relationshipType: null,
+			targetEntity: value
+		});
+	};
+
+	handleRelationshipTypeChange = (value: _Relationship) => {
+		this.setState({
+			relationship: value,
+			relationshipType: value.relationshipType
+		});
+	};
+
+	handleSave = () => {
+		const {onSave} = this.props;
+		if (onSave) {
+			if (this.state.relationship) {
+				onSave(this.state.relationship);
+			}
+		}
+	};
+
+	calculateProgressAmount() {
+		if (!this.state.entityType) {
+			return 1;
+		}
+
+		if (!this.state.targetEntity) {
+			return 30;
+		}
+
+		if (!this.state.relationshipType) {
+			return 60;
+		}
+
+		return 100;
+	}
+
+	renderEntityTypeSelect() {
+		return (
+			<FormGroup>
+				<ControlLabel>Entity Type</ControlLabel>
+				<ReactSelect
+					name="entityType"
+					options={ENTITY_TYPE_VALUES}
+					value={this.state.entityType}
+					onChange={this.handleEntityTypeChange}
+				/>
+			</FormGroup>
+		);
+	}
+
+	renderEntitySelect() {
+		return (
+			<EntitySearchField
+				cache={false}
+				disabled={!this.state.entityType}
+				instanceId="publication"
+				label={_.get(this.state, 'entityType.label', 'Entity')}
+				name="entity"
+				type={_.get(this.state, 'entityType.id')}
+				value={this.state.targetEntity}
+				onChange={this.handleEntityChange}
+			/>
+		);
+	}
+
+	renderRelationshipSelect() {
+		const {baseEntity, relationshipTypes} = this.props;
+
+		const otherEntity = {
+			bbid: _.get(this.state, ['targetEntity', 'id']),
+			defaultAlias: {
+				name: _.get(this.state, ['targetEntity', 'text'])
+			},
+			type: _.get(this.state, ['targetEntity', 'type'])
+		};
+
+		const relationships = generateRelationshipSelection(
+			relationshipTypes, baseEntity, otherEntity
+		);
+
+		return (
+			<FormGroup>
+				<ControlLabel>Relationship</ControlLabel>
+				<ReactSelect
+					disabled={!this.state.targetEntity}
+					name="relationshipType"
+					optionRenderer={Relationship}
+					options={relationships}
+					value={this.state.relationship}
+					valueKey="selectId"
+					valueRenderer={Relationship}
+					onChange={this.handleRelationshipTypeChange}
+				/>
+			</FormGroup>
+		);
+	}
+
+	render() {
+		const {onCancel, onClose, baseEntity} = this.props;
+
+		const submitDisabled = this.calculateProgressAmount() < 100;
+
+		return (
+			<Modal show bsSize="large" onHide={onClose}>
+				<Modal.Header>
+					<Modal.Title>Add a relationship</Modal.Title>
+				</Modal.Header>
+				<Modal.Body>
+					<p>
+						<strong>
+							Use this form to add links between this{' '}
+							{baseEntity.type}
+							{' '}and other entities.
+						</strong>
+						{' '}For example, you can link a creator
+						to a work as an author, or a work to another work
+						to show translation or derivation.
+					</p>
+					<hr/>
+					<Row>
+						<Col md={10} mdOffset={1}>
+							<form>
+								<Row>
+									<Col md={4}>
+										{this.renderEntityTypeSelect()}
```

Comment:
I suggest reorganizing the editor modal in this way

1. Remove EntityTypeSelect
2. Amend EntitySelect so that a user can search for any entity, limited to those that the current entity can have a valid relationship with
  - For example, a Work doesn't seem to have any allowable relationship with a Publication, so Publications should be filtered out. I also imagine we don't want Areas appearing there either.
  - The search results can be arranged by entity type, under an entity type header in the select
3. Keep RelationshipSelect as is, showing the available relationships